apiserver/facades/client/sshclient/facade_test.go

```diff
@@ -504,15 +504,186 @@ func (s *facadeSuite) TestGetVirtualHostnameForEntity(c *gc.C) {
 			c.Assert(err, jc.ErrorIsNil)
 			c.Assert(res.Address, gc.Equals, t.expected)
 		}
-
 	}
 }
 
+func (s *facadeSuite) TestHostKeyForTarget(c *gc.C) {
+	ctrl := gomock.NewController(c)
+	defer ctrl.Finish()
+	backend := mocks.NewMockBackend(ctrl)
+	authorizer := mocks.NewMockAuthorizer(ctrl)
+	broker := mocks.NewMockBroker(ctrl)
+
+	authorizer.EXPECT().AuthClient().Return(true).Times(1)
+
+	facade, err := sshclient.InternalFacade(backend, nil, authorizer, s.callContext,
+		func(context.Context, environs.OpenParams) (sshclient.Broker, error) {
+			return broker, nil
+		},
+	)
+	c.Assert(err, gc.IsNil)
+
+	// Set superuser for all calls.
+	controllerTag := names.NewControllerTag("controller00-0000-0000-000000000000")
+	backend.EXPECT().ControllerTag().Return(controllerTag).AnyTimes()
+	authorizer.EXPECT().HasPermission(permission.SuperuserAccess, controllerTag).AnyTimes()
+
+	// Test container target hits correct methods.
+	gomock.InOrder(
+		backend.EXPECT().UnitVirtualPublicHostKeyPEM("postgresql/1").Return("postgres-container-host-key", nil).Times(1),
+		backend.EXPECT().SSHServerHostKey().Return("server-host-key", nil).Times(1),
+	)
+	res := facade.PublicHostKeyForTarget(params.SSHHostKeyRequestArg{
+		Hostname: "charm.1.postgresql.8419cd78-4993-4c3a-928e-c646226beeee.juju.local",
+	})
+	c.Assert(res, gc.Equals, params.SSHHostKeyResult{
+		Error:             nil,
+		HostKey:           "postgres-container-host-key",
+		JumpServerHostKey: "server-host-key",
+	})
+
+	// Test unit target hits correct methods.
+	gomock.InOrder(
+		backend.EXPECT().UnitVirtualPublicHostKeyPEM("openfga/1").Return("openfga-container-host-key", nil).Times(1),
+		backend.EXPECT().SSHServerHostKey().Return("server-host-key", nil).Times(1),
+	)
+	res = facade.PublicHostKeyForTarget(params.SSHHostKeyRequestArg{
+		Hostname: "1.openfga.8419cd78-4993-4c3a-928e-c646226beeee.juju.local",
+	})
+	c.Assert(res, gc.Equals, params.SSHHostKeyResult{
+		Error:             nil,
+		HostKey:           "openfga-container-host-key",
+		JumpServerHostKey: "server-host-key",
+	})
+
+	// Test machine target hits correct methods.
+	gomock.InOrder(
+		backend.EXPECT().MachineVirtualPublicHostKeyPEM("1").Return("machine-host-key", nil).Times(1),
+		backend.EXPECT().SSHServerHostKey().Return("server-host-key", nil).Times(1),
+	)
+	res = facade.PublicHostKeyForTarget(params.SSHHostKeyRequestArg{
+		Hostname: "1.8419cd78-4993-4c3a-928e-c646226beeee.juju.local",
+	})
+	c.Assert(res, gc.Equals, params.SSHHostKeyResult{
+		Error:             nil,
+		HostKey:           "machine-host-key",
+		JumpServerHostKey: "server-host-key",
+	})
+}
+
+func (s *facadeSuite) TestHostKeyForTargetUnhappyPaths(c *gc.C) {
```

Comment:
Instead of unhappy paths, maybe `WithError`? And what do you think about a table test, so that each error has a clear "description".

Reply:
I'm really not a fan of table tests so I'd prefer to leave like this (Unless we absolutely won't let it through without table tests?). Renamed to TestPublicHostKeyForTargetErrors .